### PR TITLE
Model checks

### PR DESCRIPTION
**Pull request by znarf**
We have some consistency issues in the database and this is an attempt at documenting them and their fix.

### CODE REVIEW

File: checks/hosted-collectives.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     AND "isHostAccount" IS FALSE
+     AND "ParentCollectiveId" IS NULL
+     AND "HostCollectiveId" IS NOT NULL
+     AND "type" NOT IN ('ORGANIZATION', 'USER')`,
+    { type: sequelize.QueryTypes.SELECT, raw: true },
+  );
+
+  if (results[0].count > 0) {
+    if (check) {
+      throw new Error('Hosted Collectives without hostFeePercent');
```
Comment (Betree) [inline checks/hosted-collectives.js:25]: Is that an issue, given that we default to the host's hostFeePercent?

File: checks/transactions.js
Comment (Betree) [inline checks/transactions.js:1]: Given what we said about refunds the other day, we may want to check transactions that have a deleted payment method in the future.

**Review by Betree — COMMENTED** [review]
I was looking for this today in the codebase, it would be great to see it merged. Maybe we can move the code to `scripts/checks` for consistency.

**Comment by znarf** [comment]
@Betree Let's try to pair a bit to complete a first version.

File: checks/users.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+     LEFT JOIN "Collectives" c
+     ON c."id" = u."CollectiveId"
+     WHERE u."deletedAt" IS NULL
+     AND (c."deletedAt" IS NOT NULL OR c."id" IS NULL)`,
+    { type: sequelize.QueryTypes.SELECT, raw: true },
+  );
+
+  if (results[0].count > 0) {
+    if (check) {
+      throw new Error('No Users without a matching Collective');
```
Comment (Betree) [inline checks/users.js:22]: One closely related that I wanted to check today is cases of `Users` entries linked to non-`USER` Collectives. We have a few:

```sql
select c.slug, c.type
FROM "Users" u
INNER JOIN "Collectives" c ON u."CollectiveId" = c.id
WHERE c.type != 'USER'
AND c."deletedAt" IS NULL
AND u."deletedAt" IS NULL
```

**Comment by Betree** [comment]
Today I investigated a case where, following manual edits to the PayPal account, OCNZ removed the PayPal webhook that lets us record transactions asynchronously. Thankfully we have a reconciliation script, but this is very problematic and could be easily detected using a similar approach to what's done here.

I would advocate to enlarge the scope of this work to not just "DB checks", but "Sanity checks". For PayPal, I would like to run a job that checks that:
- All hosts connected PayPal credentials are still valid (they should return a 200 for a simple request)
- All hosts with PayPal configured have a valid webhook on their accounts ([validation code already exists](https://github.com/opencollective/opencollective-api/blob/6ec8263b3ccbddec2fc0fb5906e0405efb78159a/server/lib/paypal.ts))

**Comment by znarf** [comment]
> Maybe we can move the code to scripts/checks for consistency.

I will keep it in this new `checks` folder for now as I want to push this as a concept different from scripts.

**Review by Betree — APPROVED** [review]
Looking good! The folder pattern is future-proof enough, we'll be able to add PayPal checks in `checks/paypal` and so on.

**Pull request merged**